packages/graphql/src/translate/translate-read-new.ts

```diff
@@ -0,0 +1,201 @@
+        node,
+        context,
+        fieldsByTypeName,
+        varName,
+    });
+    [projStr] = projection;
+    cypherParams = { ...cypherParams, ...projection[1] };
+    if (projection[2]?.authValidateStrs?.length) {
+        projAuth = `CALL apoc.util.validate(NOT(${projection[2].authValidateStrs.join(
+            " AND "
+        )}), "${AUTH_FORBIDDEN_ERROR}", [0])`;
+    }
+
+    if (projection[2]?.connectionFields?.length) {
+        projection[2].connectionFields.forEach((connectionResolveTree) => {
+            const connectionField = node.connectionFields.find(
+                (x) => x.fieldName === connectionResolveTree.name
+            ) as ConnectionField;
+            const connection = createConnectionAndParams({
+                resolveTree: connectionResolveTree,
+                field: connectionField,
+                context,
+                nodeVariable: varName,
+            });
+            connectionStrs.push(connection[0]);
+            cypherParams = { ...cypherParams, ...connection[1] };
+        });
+    }
+
+    if (whereInput) {
+        const where = createWhereAndParams({
+            whereInput,
+            varName,
+            node,
+            context,
+            recursing: true,
+        });
+        if (where[0]) {
+            whereStrs.push(where[0]);
+            cypherParams = { ...cypherParams, ...where[1] };
+        }
+    }
+
+    const whereAuth = createAuthAndParams({
+        operation: "READ",
+        entity: node,
+        context,
+        where: { varName, node },
+    });
+    if (whereAuth[0]) {
+        whereStrs.push(whereAuth[0]);
+        cypherParams = { ...cypherParams, ...whereAuth[1] };
+    }
+
+    if (whereStrs.length) {
+        whereStr = `WHERE ${whereStrs.join(" AND ")}`;
+    }
+
+    const allowAndParams = createAuthAndParams({
+        operation: "READ",
+        entity: node,
+        context,
+        allow: {
+            parentNode: node,
+            varName,
+        },
+    });
+    if (allowAndParams[0]) {
+        cypherParams = { ...cypherParams, ...allowAndParams[1] };
+        authStr = `CALL apoc.util.validate(NOT(${allowAndParams[0]}), "${AUTH_FORBIDDEN_ERROR}", [0])`;
+    }
+
+    if (optionsInput) {
+        if (hasOffset) {
+            offsetStr = `SKIP $${varName}_offset`;
+            cypherParams[`${varName}_offset`] = optionsInput.offset;
+        }
+
+        if (hasLimit) {
+            limitStr = `LIMIT $${varName}_limit`;
+            cypherParams[`${varName}_limit`] = optionsInput.limit;
+        }
+
+        if (optionsInput.sort && optionsInput.sort.length) {
+            if (hasLimit) {
+                const sortFields = optionsInput.sort.map((sort) => Object.keys(sort)[0]);
+                const orderByProjection = createProjectionAndParams({
+                    node,
+                    context,
+                    fieldsByTypeName: {
+                        [node.name]: Object.fromEntries(
```

Comment:
`Object.fromEntries` required me to bump the ts target (from es2015?) I think.  There may be polyfills available for older targets.

Reply:
The polyfill for this is simple to implement if it's beneficial to maintain older target language support: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries#polyfill